### PR TITLE
test/crimson: implement a remote async TestPeer for crimson msgr tests

### PR DESCRIPTION
**Pull request by cyx1231st**
To do the same set of crimson msgr v2 protocol tests against a remote async msgr TestPeer, run:
```
$./test_seastar_messenger_peer
$./unittest_seastar_messenger --v2-testpeer-islocal=0
```
This verifies all the failover/racing/policy behaviors of v2 crimson msgr (implemented in `test_v2_protocol()`, see #30162, #30313, #30803, #30925) are compatible with v2 async msgr.

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`

</details>

### CODE REVIEW

File: src/test/crimson/test_messenger_peer.cc [hunk 1]
```diff
@@ -0,0 +1,434 @@
...
+#include "messages/MCommandReply.h"
+#include "messages/MOSDOp.h"
+#include "msg/Dispatcher.h"
+#include "msg/Messenger.h"
+
+#include "test_cmds.h"
+
+namespace {
+
+#define dout_subsys ceph_subsys_ms
```
Comment (tchaikov) [inline src/test/crimson/test_messenger_peer.cc]: s/ms/test/

we've already have a dedicated subsys for test. see 

https://github.com/ceph/ceph/blob/190b0bf433b08682f22aa2914a808dfd8976cf25/src/common/subsys.h#L81

File: src/test/crimson/test_messenger_peer.cc [hunk 2]
```diff
@@ -0,0 +1,434 @@
...
+
+  Connection* tracked_conn = nullptr;
+  unsigned pending_send = 0;
+
+  bool ms_can_fast_dispatch_any() const override { return true; }
+  bool ms_can_fast_dispatch(const Message* m) const override { return true; }
+  void ms_fast_dispatch(Message* m) override {
+    auto conn = m->get_connection().get();
+    if (tracked_conn == nullptr) {
+      lderr(cct) << "[!TestPeer] got op from Test(conn "
```
Comment (tchaikov) [inline src/test/crimson/test_messenger_peer.cc]: please avoid using `lderr()` for non-error messages.

File: src/test/crimson/test_cmds.h [hunk 3]
```diff
@@ -0,0 +1,76 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
```
Comment (tchaikov) [inline src/test/crimson/test_cmds.h]: also, could you transpose "crimson" and "test" in the title? so it reads

> test/crimson: extract msgr test commands to header

?

File: src/test/crimson/test_messenger_peer.cc (hunk 1 above)
Comment (cyx1231st) [reply src/test/crimson/test_messenger_peer.cc]: fixed

File: src/test/crimson/test_messenger_peer.cc (hunk 2 above)
Comment (cyx1231st) [reply src/test/crimson/test_messenger_peer.cc]: changed to `ldout(cct, 0)`

File: src/test/crimson/test_cmds.h (hunk 3 above)
Comment (cyx1231st) [reply src/test/crimson/test_cmds.h]: sure, fixed.

**Pull request merged**